synapse/handlers/deactivate_account.py

```diff
@@ -40,7 +40,8 @@ def __init__(self, hs):
 
         # Start the user parter loop so it can resume parting users from rooms where
         # it left off (if it has work left to do).
-        hs.get_reactor().callWhenRunning(self._start_user_parting)
+        if hs.config.worker_app is None:
+            hs.get_reactor().callWhenRunning(self._start_user_parting)
```

Comment:
I think it is worth pointing out also that we don't seem to use this pattern of `callWhenRunning` anywhere else for background processes. I wonder if that method can just go away and we can directly call the code from it?

Not really related to this PR, just found it odd.

Reply:
I think we tend to do loopings calls, which also won't run until we're started? I think it makes sense to only start work once we've finished starting up